### PR TITLE
Adding `api-inference-community` to `huggingface_hub`.

### PR DESCRIPTION
**Pull request by Narsil**
- The idea is to make all external contributions centralized.
- Moved the quality checks (could make them uniform later)
- `api-inference-community` should be relatively independant from `huggingface_hub` still.

### CODE REVIEW

**Comment by julien-c** [comment]
Thanks! Looks good to me but adding @LysandreJik and @n1t0 to the party

**Comment by julien-c** [comment]
Does the CI take a long time? If it does, maybe we'll want to restrict the Inference-API's CI to changes to that folder.

**Review by LysandreJik — APPROVED** [review]
Cool, looking forward to having everything centralized.

**Comment by Narsil** [comment]
Not sure how easy to run conditional actions but if it's easy then yes let's do it !

Otherwise the CI is super fast as we only run the package CI, everything else is considered "slow" as it would require model downloads (and I don't want to force users to test against dummy models as it is a bit more complex to do).

Before deployment, we run the `test_dockers` slow CI which actually build the container + run the test suite on those with a live model. We can do that framework per framework making it livable (running tests against all frameworks all the time seems a bit like a waste since we're relying on docker to make sure no unforeseen change happens within the API)

Conditional full docker tests would be the best so that PR creators can be alerted early if something doesn't fit.

File: .github/workflows/python-api-quality.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Check code quality
```
Comment (julien-c) [inline .github/workflows/python-api-quality.yaml:1]:
```suggestion
name: Inference API code quality
```

File: .github/workflows/python-api-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: Python-tests
```
Comment (julien-c) [inline .github/workflows/python-api-tests.yaml:1]:
```suggestion
name: Inference API Python-tests
```

File: .github/workflows/python-api-quality.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+on: push
```
Comment (julien-c) [inline .github/workflows/python-api-quality.yaml:3]:
```suggestion
on:
  push:
    paths:
      - "api-inference-community/**"
```

File: .github/workflows/python-api-tests.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+on:
+  push:
+    branches:
+      - "*"
```
Comment (julien-c) [inline .github/workflows/python-api-tests.yaml:6]:
```suggestion
on:
  push:
    paths:
      - "api-inference-community/**"
```

File: README.md [hunk 5]
```diff
@@ -140,6 +140,10 @@ We add a custom "Use in Asteroid" button.
 
 When clicked you get a library-specific code sample that you'll be able to specify. 🔥
 
+## API integration into the huggingface.co hub
```
Comment (julien-c) [inline README.md:143]:
```suggestion
## Inference API integration into the huggingface.co hub
```

File: README.md [hunk 6]
```diff
@@ -140,6 +140,10 @@ We add a custom "Use in Asteroid" button.
...
+
+In order to get functional widgets on the hub for your models check out this [doc](https://github.com/huggingface/huggingface_hub/tree/master/api-inference-community)
```
Comment (julien-c) [inline README.md:145]:
```suggestion
In order to get a functional Inference API on the hub for your models (and thus, cool working widgets!) check out this [doc](https://github.com/huggingface/huggingface_hub/tree/master/api-inference-community)
```

**Review by julien-c — CHANGES_REQUESTED** [review]

**Comment by Narsil** [comment]
Hi @julien-c do you think the changes made are OK?

I added a simple docker test only for allennlp (afaik, I would need one for each framework, because we can't have multiple `on` clauses in a single yaml file).

Also I am not 100% sure about this docker because it does not seem to run on this PR, is that normal ? I though GH actions should run on current PR, but maybe I'm wrong.

**Comment by julien-c** [comment]
not sure why the CI suite doesn't run, maybe it will after we merge this first PR? or maybe add a `pull_request:` event clause with the same paths?

**Review by LysandreJik — APPROVED** [review]
LGTM, thanks @Narsil!

I'm pretty sure it's normal that the CI doesn't run as long as this isn't on `master`, we had the same issue on transformers a while back.

**Review by julien-c — APPROVED** [review]

**Pull request merged**